rustls/src/builder.rs

```diff
@@ -1,62 +1,55 @@
-use crate::crypto::{CryptoProvider, SupportedKxGroup};
+use crate::crypto::CryptoProvider;
 use crate::error::Error;
-use crate::suites::SupportedCipherSuite;
 use crate::versions;
 
 use alloc::format;
-use alloc::vec::Vec;
 use core::fmt;
 use core::marker::PhantomData;
+use std::sync::Arc;
 
 /// A [builder] for [`ServerConfig`] or [`ClientConfig`] values.
 ///
 /// To get one of these, call [`ServerConfig::builder()`] or [`ClientConfig::builder()`].
 ///
 /// To build a config, you must make at least three decisions (in order):
 ///
-/// - Which protocol primitives should be supported (cipher suites, key exchange groups, protocol versions)?
+/// - Which cryptographic provider to use, such as [`ring::provider`]?
```

Comment:
The new API reduces the number of "must decides." In particular the choice of protocol primitives went away. Now I think it's more accurate to say:

```
/// To build a config, you must make at least two decisions (in order):
///
/// - How should this client or server verify certificates provided by its peer?
/// - What certificates should this client or server present to its peer?
```

To some extent you're "making a decision" by choosing `ClientConfig::builder` vs `ClientConfig::builder_with_XXX`. But that doesn't have to be part of this list. The goal of this list is to orient people to "what are the must-call methods I use to get to the point where I can finish building a config."

Reply:
Fixed up. Thanks!